run/batch_applier.go

```diff
@@ -197,7 +199,7 @@ func (a *BatchApplier) apply(path string, options *ApplyOptions) (*ApplyAttempt,
 	finish := a.Clock.Now()
 
 	appliedFile := ApplyAttempt{
-		FilePath:     path,
+		FilePath:     subPath,
```

Comment:
Should we rename this field to `Namespace` or something similar? `FilePath` doesn't really represent what it is any more.

Reply:
I'm not sure about this, throughout the code the idea is that we're applying directories/paths. Namespace is something inferred from the basename of the path.

It still is a file path, it's just relative now. For example, the list of failed apply tasks is constructed from `ApplyAttempt.FilePath` and to apply them again you `path.Join` the root path with it. [This](https://github.com/utilitywarehouse/kube-applier/blob/master/run/runner.go#L189-L192) would feel wrong if it read `r.lastRunFailures[i] = f.Namespace` and consequently [here](https://github.com/utilitywarehouse/kube-applier/blob/master/run/runner.go#L104) `dirs` would be a list of namespaces.

In practice, namespace and directory are interchangeable (by design) but it _feels_ somewhat wrong to call it a namespace here.

Reply:
That makes sense.